# Assignments

# Question 1:

# Write a program to input T numbers(N) from user and print count of digits of the given numbers.



**Input :**

2

100

10101

**Output :**

3

5

![Smart](https://i.imgflip.com/1j8x99.jpg)

In [2]:
def main():
    # YOUR CODE GOES HERE
    # Please take input and print output to standard input/output (stdin/stdout)
    # E.g. 'input()/raw_input()' for input & 'print' for output
    operation=int(input())
    for i in range(operation):
        num = int(input())
        count = len(str(num))
        print(count)


    return 0

if __name__ == '__main__':
    main()

5
1213
4
1254
4
1256
4
21
2
235
3


# Question 2:

# Write a program to input T numbers(N) from user and find reverse of the given number.

**Input :**

2

101

105

**Output :**

101

501

# Using Slicing Method- string only

In [1]:
operation = int(input())
for i in range (operation):
    num = int(input())
    new_num = str(num)
    reverse = new_num[::-1]
    print (reverse)

2
124
421
256
652


# Using floor division method- intergers permitted

![output](https://media.geeksforgeeks.org/wp-content/cdn-uploads/program-to-reverse-digits-of-a-number-1024x512.png)

Algorithm:

Input:  num

(1) Initialize rev_num = 0

(2) Loop while num > 0

     (a) Multiply rev_num by 10 and add remainder of num  
          divide by 10 to rev_num
               rev_num = rev_num*10 + num%10;
     (b) Divide num by 10
     
(3) Return rev_num

In [1]:
def main():
    # YOUR CODE GOES HERE
    # Please take input and print output to standard input/output (stdin/stdout)
    # E.g. 'input()/raw_input()' for input & 'print' for output
    operation = int(input())
    for i in range(operation):
        num = int(input())
        rev = 0
        while(num > 0):
            a = num % 10
            rev = rev * 10 + a
            num = num // 10
        print(rev)        

    return 0

if __name__ == '__main__':
    main()


2
123
321
002
2


# Question 3:

# You are given an integer N you need to print all the Prime Numbers between 1 and N.


**Prime numbers are numbers that have only 2 factors: 1 and themselves. For example, the first 5 prime numbers are 2, 3, 5, 7, and 11.**

**We already know that 1 is not a prime number, so we won't be checking for 1.**

![Shut_up](https://i.imgflip.com/4/3a7qxu.jpg)

## This program solution can be found by using function creating method at <a href="https://www.youtube.com/watch?v=c84Y8aWYDh0" target="_blank">this link</a>

# Correct_Code:

In [6]:
def main():
    # YOUR CODE GOES HERE
    # Please take input and print output to standard input/output (stdin/stdout)
    # E.g. 'input()/raw_input()' for input & 'print' for output
    def isPrime(n):
      #since 0 and 1 is not prime return false.
      if(n==1 or n==0):
        return False

      #Run a loop from 2 to n/2
      for i in range(2,(n//2)+1):
        #if the number is divisible by i, then n is not a prime number.
        if(n%i==0):
          return False

      #otherwise, n is prime number.
      return True



    # Driver code
    N = int(input());
    #check for every number from 1 to N
    for i in range(1,N+1):
      #check if current number is prime
      if(isPrime(i)):
        print(i,end="\n")
                    
 
    return 0

if __name__ == '__main__':
    main()

5
2
3
5


# Let's create a function:


In [6]:
def isprime(digit):

    num = int(digit)
    if num == 1:
        #print(str(num) + " is not a prime number")
        pass
    elif num == 2:
        #print(str(num) + " is a prime number")
        return 1
    else:    
        for i in range(2,round(num/2)+1):
            if num % 2 == 0:
                #print(str(num) + " is not a prime number")
                break
            elif num % 2 != 0:
                if i == round(num/2):
                    #print(str(num) + " is a prime number")
                    return 1


In [11]:
def primelistmaker(number):
    n = int(number)
    y = 1
    counter = 0
    yes_no_prime = 0
    
    list_of_primes = []

    while counter <= n-1:
        yes_no_prime = isprime(y)
        if yes_no_prime == 1:
            list_of_primes.append(y)
            counter +=1
        y +=1  
    print("The list of first "+ str(n) + " prime numbers are:")
    print(list_of_primes)
    print(len(list_of_primes))
 

In [12]:
number = input("Enter the number here: ")
primelistmaker(number)

Enter the number here: 10
The list of first 10 prime numbers are:
[2, 3, 5, 7, 9, 11, 13, 15, 17, 19]
10


# Trial Program: Print First N Prime Numbers (including functions)

In [39]:
def isprime(digit):

    num = int(digit)
    if num == 1:
        #print(str(num) + " is not a prime number")
        pass
    elif num == 2:
        #print(str(num) + " is a prime number")
        return 1
    else:    
        for i in range(2,round(num/2)+1):
            if num % 2 == 0:
                #print(str(num) + " is not a prime number")
                break
            elif num % 2 != 0:
                if i == round(num/2):
                    #print(str(num) + " is a prime number")
                    return 1
                
def primelistmaker(number):
    
    n = int(number)
    y = 1
    counter = 0
    yes_no_prime = 0
    
    list_of_primes = []

    while counter <= n:
        yes_no_prime = isprime(y)
        if yes_no_prime == 1:
            list_of_primes.append(y)
            counter +=1
        y +=1    
    print("The list of first "+ str(n) + " prime numbers are:")

    print(*list_of_primes, sep ="\n")
    
           

In [40]:
number = input("Enter the number here: ")
primelistmaker(number)

Enter the number here: 5
The list of first 5 prime numbers are:
2
3
5
7
9
11


# Program to print primes till Nth number :


In [1]:
def isprime(digit):

    num = int(digit)
    if num == 1:
        #print(str(num) + " is not a prime number")
        pass
    elif num == 2:
        #print(str(num) + " is a prime number")
        return 1
    else:    
        for i in range(2,round(num/2)+1):
            if num % 2 == 0:
                #print(str(num) + " is not a prime number")
                break
            elif num % 2 != 0:
                if i == round(num/2):
                    #print(str(num) + " is a prime number")
                    return 1
                
def primelistmaker(number):
    
    n = int(number)
    y = 1
    counter = 0
    yes_no_prime = 0
    
    list_of_primes = []

    while y <= n:
        yes_no_prime = isprime(y)
        if yes_no_prime == 1:
            list_of_primes.append(y)
            #counter +=1
        y +=1    
    print("The list of first "+ str(n) + " prime numbers are:")

    print(*list_of_primes, sep ="\n")
               

In [2]:
number = input("Enter the number here: ")
primelistmaker(number)

Enter the number here: 11
The list of first 11 prime numbers are:
2
3
5
7
9
11


In [5]:
def isPrime(n):
  #since 0 and 1 is not prime return false.
  if(n==1 or n==0):
    return False
   
  #Run a loop from 2 to n/2
  for i in range(2,(n//2)+1):
    #if the number is divisible by i, then n is not a prime number.
    if(n%i==0):
      return False
   
  #otherwise, n is prime number.
  return True
 
 
 
# Driver code
N = int(input());
#check for every number from 1 to N
for i in range(1,N+1):
  #check if current number is prime
  if(isPrime(i)):
    print(i,end="\n")

5
2
3
5


# Homework Problems:

### Question 1:
## Given an integer A, determine whether it is palindromic or not.

**A palindrome integer is an integer X for which reverse(X) = X where reverse(X) is X with its digits reversed. For e.g., reverse(123) = 321.**

**What is a Palindrome?**

* *A palindrome is nothing but any number or a string which remains unaltered when reversed.*

**Example:** 12321

**Output:** Yes, a Palindrome number

**Example:** RACECAR

**Output:** Yes, a Palindrome string

# String_Method_Code:

In [5]:
string = input(("Enter a string:"))
if(string==string[::-1]):
      print("The string is a palindrome")
else:
      print("Not a palindrome")

Enter a string:RACECAR
The string is a palindrome


# Integer Method_Correct Code: 

In [25]:
num=int(input("Enter a number:"))
temp=num
rev=0
while(num>0):
    dig=num%10
    rev=rev*10+dig
    num=num//10
if(temp==rev):
    print("The number is palindrome!")
else:
    print("Not a palindrome!")

Enter a number:121
The number is palindrome!


In [44]:
num = int(input())
temp = num
rev = 0

while (num>0):
    # This will return the remainder and give the unit digit
    digit = num % 10
    
    # This will help us in reverse the unit digit 
    rev = rev*10 + digit
    
    # This will give us the floor divison Quotient 
    num = num // 10
    
if (temp == rev):
    print("Yes Palindrome")
else:
    print("NO Palindrome")

121
Yes Palindrome


# Palindrome Program using Function: 

## This program solution can be found by using function creating method at <a href="https://www.youtube.com/watch?v=yubRKwixN-U" target="_blank">this link</a>

In [11]:
def ispalindrome(num):
    num = int()
    if num < 0 : return False
    
    div = 1
    while num >= 10 * div:
        div*=10
    while num:
        right = num % 10
        left = num // div
        
        if left != right: 
            print("Not Palindrome")
        num = (num % div) // 10
        div = div / 100
    print("Yes Palindrome")
num = input()
ispalindrome(num)

121
Yes Palindrome


### Question 2:

# You are given two integers A and B. You have to find the value of AB.

#### NOTE: The value of answer is always less than or equal to 109.

### Input_Method - I

In [7]:
a,b = input().split(',')
a,b = int(a),int(b)
print(pow(a,b))


2,5
32


### Input_Method - II


In [8]:
a = int(input())
b = int(input())
print(pow(a,b))


2
5
32


## Question 3:

### Given a month number, A and an year, B, return the number of days in that month (take leap years into account).

**Input 1:**

 A = 2
 
 B = 2020
 
**Input 2:**

 A = 10
 
 B = 1999

**Output 1:**

 29

**Output 2:**

 31

![Dexter](https://i.kym-cdn.com/entries/icons/facebook/000/015/487/failure.jpg)

### How to Check Whether a Year is a Leap Year or Not?

* A leap year has 366 days (the extra day is the 29th of February), and it comes after every four years. 
* To check if a year is a leap year, divide the year by 4. 
* If it is fully divisible by 4, it is a leap year. For example, the year 2016 is divisible 4, so it is a leap year, whereas, 2015 is not.

**Note: However, Century years like 300, 700, 1900, 2000 need to be divided by 400 to check whether they are leap years or not.**

# Using Dictionary: Wrong_code

In [9]:
common_year = {1:{'Jan':[31]},2:{'Feb':[28]},3:{'Mar':[31]},4:{'Apr':[30]},5:{'May':[31]},6:{'Jun':[30]},7:{'Jul':[31]},8:{'Aug':[31]},9:{'Sept':[30]},10:{'Oct':[31]},11:{'Nov':[30]},12:{'Dec':[31]}
leap_year = {1:{'Jan':[31]},2:{'Feb':[29]},3:{'Mar':[31]},4:{'Apr':[30]},5:{'May':[31]},6:{'Jun':[30]},7:{'Jul':[31]},8:{'Aug':[31]},9:{'Sept':[30]},10:{'Oct':[31]},11:{'Nov':[30]},12:{'Dec':[31]}
  
    
A = int(input())
for i in range (1,13):
    B = int (input())
    if B % 4 == 0:
        if (B % 400 == 0):
            print("It's a Leap year: ",B)
            print ("Number of days are: ")
            for i in leap_year :
                print(i,leap_year[i])
        else:
            print("It's a Common Year: ",B)
            print ("Number of days are: ")
            for i in common_year :
                print(i,common_year[i])
                
    elif (B % 100 == 0):
        print("It's a Common Year: ",B)
        print ("Number of days are: ")
        for i in common_year :
            print(i,common_year[i])
        
    else:
        pass

SyntaxError: invalid syntax (Temp/ipykernel_41196/3336795272.py, line 2)

# Corrent_Code

In [11]:
month = int(input())
year = int(input())
if month in (1,3,5,7,8,10,12):
    no_days = 31
    print("Number of days in %d is %d " %(month,no_days))
elif month in (4,6,9,11):
    no_days = 30
    print("Number of days in %d is %d" %(month,no_days))
elif month ==2:
    if (year % 4 == 0) and (not(year%100==0) or (year%400 == 0)):
        no_days=29
    else:
        no_days=28
        print("Number of days in %d is %d" %(month,no_days))
else:
    print("Invalid Entry")

12
1999
Number of days in 12 is 31 


In [14]:

        A = int(input())
        B = int(input())
        
        if A in (1,3,5,7,8,10,12):
            no_days = 31
            print(no_days)
        elif A in (4,6,9,11):
            no_days = 30
            print(no_days)
        elif A == 2:
            if (B % 4 == 0) and (not(B % 100 == 0) or (B % 400 == 0)):
                no_days=29
                print(no_days)
            else:
                no_days=28
                print(no_days)
        else:
            print("Invalid Entry")
            


2
1996
29


# scaler_correct_code

In [15]:
class Solution:
    # @param A : integer
    # @param B : integer
    # @return an integer
    def solve(self, A, B):
        

        if A in (1,3,5,7,8,10,12):
            no_days = 31
            return no_days
        elif A in (4,6,9,11):
            no_days = 30
            return no_days
        elif A == 2:
            if (B % 4 == 0) and (not(B % 100 == 0) or (B % 400 == 0)):
                no_days=29
                return no_days
            else:
                no_days=28
                return no_days
        else:
            pass       



3
1994
31


# Question 4:

## You are given a lowercase latin alphabetic character C. You have to tell whether it is a vowel or not.

### The characters 'a', 'e', 'i', 'o', and 'u' are called vowels.

In [19]:
letter = input()    # Allowing only the lowercase 

if letter in ('a','e','i','o','u'):
    print(letter,"is a vowel")
else:
    print(letter,"is a consonant (not a vowel)")

A
A is a consonant (not a vowel)


# scaler_correct_code

In [ ]:
def isvowel(C):
    ans = None
    # YOUR CODE GOES HERE
    if C in ('a','e','i','o','u'):
        return 1
    else:
        return 0

    
    
    
    return ans

# Question 5:

## Write a program to input from user five numbers(A, B, C, D & E) representing marks of student in 5 subjects out of 100. You have to calculate percentage and then Grade of each student.

* If percentage >= 90% : Grade A
* If percentage >= 80% : Grade B
* If percentage >= 70% : Grade C
* If percentage >= 60% : Grade D
* If percentage >= 40% : Grade E
* If percentage < 40% : Grade F

#### NOTE: You have to take the lowest integer of percentage. E.g. 90.8% will be treated as 90%.

In [14]:
def main():
    # YOUR CODE GOES HERE
    # Please take input and print output to standard input/output (stdin/stdout)
    # E.g. 'input()/raw_input()' for input & 'print' for output
    A,B,C,D,E=input().split(' ')
    A,B,C,D,E=int(A),int(B),int(C),int(E),int(E)
    total = A+B+C+D+E
    perct = round((total/500)*100) 

    if perct >= 90:
        print(perct, "A")
    elif 90 > perct >= 80:
        print(perct, "B")
    elif 80 > perct >= 70:
        print(perct, "C")
    elif 70 > perct >= 60:
        print(perct, "D") 
    elif 60 > perct >= 40:
        print(perct, "E") 
    elif perct < 40:
        print(perct, "F")
    else:
        pass


    return 0

if __name__ == '__main__':
    main()

50 60 70 80 90
72 C
